OpenMP: Refactor parallel_reduce to share code with parallel_for

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60185 Ensure num_threads is initialized before calling omp_get_max_threads
* **#60184 OpenMP: Refactor parallel_reduce to share code with parallel_for**
* #60183 Ensure thread id is valid in nested parallel regions

Differential Revision: [D29287817](https://our.internmc.facebook.com/intern/diff/D29287817)

peterbell10 committed:
OpenMP: Refactor parallel_reduce to share code with parallel_for

[ghstack-poisoned]

peterbell10 committed:
Update on "OpenMP: Refactor parallel_reduce to share code with parallel_for"




[ghstack-poisoned]

peterbell10 committed:
Update on "OpenMP: Refactor parallel_reduce to share code with parallel_for"




[ghstack-poisoned]

peterbell10 committed:
Update on "OpenMP: Refactor parallel_reduce to share code with parallel_for"




[ghstack-poisoned]

peterbell10 committed:
Update on "OpenMP: Refactor parallel_reduce to share code with parallel_for"




[ghstack-poisoned]

peterbell10 committed:
Update on "OpenMP: Refactor parallel_reduce to share code with parallel_for"




[ghstack-poisoned]

peterbell10 committed:
Update on "OpenMP: Refactor parallel_reduce to share code with parallel_for"




[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29287817).

peterbell10 committed:
Update on "OpenMP: Refactor parallel_reduce to share code with parallel_for"


Differential Revision: [D29287817](https://our.internmc.facebook.com/intern/diff/D29287817)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29287817).
